error-codes/README.md

```diff
@@ -0,0 +1,97 @@
+# Error Codes
+
+## Introduction
+In the event that a node detects violation of a protocol or is unable to
+complete the necessary steps required for the protocol, it's useful to provide a
+reason for disconnection to the other end. This error code can be sent on both
+Connection Close and Stream Reset. Its purpose is similar to http response
+status. A server, on receiving an invalid request can reset the stream providing
+a `BAD_REQUEST` error code, when it's busy handling too many requests can
+provide a `RATE_LIMITED` error code, etc. An error code doesn't always indicate
+an error condition. For example, a connection may be closed prematurely because
+a connection to the same peer on a better transport is available. 
```

Comment:
I think this reads more naturally as-is.  If you add the `of` you'd also need to move the `is available` bit so it'd be something like:

> ..closed prematurely because of the existence of a connection to the same peer over a better transport.

..which to me reads clunkier than the original since now we have jammed the adjective between two prepositions and the noun is dangling off the end.

That said, `over a better transport` might be better than `on a better transport` but it's the tiniest of tiny nits.